### PR TITLE
stage2 ARM: more instructions, return values, parameters

### PR DESCRIPTION
**Pull request by joachimschmidt557**
Adding some more stuff to stage2 ARM backend

### CODE REVIEW

**Comment by joachimschmidt557** [comment]
I'll add some testcases for basic parameters, return types and non-leaf functions. After that, I'll mark this ready for review.

File: src-self-hosted/codegen/arm.zig [hunk 1]
```diff
@@ -389,12 +429,36 @@ pub const Instruction = union(enum) {
...
                 .rd = rd.id(),
                 .offset = offset.toU12(),
-                .l = load_store,
-                .w = writeback,
-                .b = byte_word,
-                .u = up_down,
-                .p = pre_post,
-                .i = if (offset == .Immediate) 0 else 1,
+                .load_store = load_store,
+                .write_back = if (write_back) 1 else 0,
```
Comment (Vexu) [inline src-self-hosted/codegen/arm.zig:433]:
```suggestion
                .write_back = @boolToInt(write_back),
```

Comment (joachimschmidt557) [reply src-self-hosted/codegen/arm.zig:433]: I was thinking of that as well. Sometimes, the bits represent the intuition, sometimes they are the opposite of the intuition:

https://github.com/ziglang/zig/blob/99a9590a165beb2f73e2644fa9bf43fce984fbcd/src-self-hosted/codegen/arm.zig#L405

https://github.com/ziglang/zig/blob/99a9590a165beb2f73e2644fa9bf43fce984fbcd/src-self-hosted/codegen/arm.zig#L437

In order to be consistent, should I use `@boolToInt` everywhere and invert the argument where necessary?

Comment (Vexu) [reply src-self-hosted/codegen/arm.zig:433]: I don't think `if (offset == .Immediate) 0 else 1` is any clearer than `@boolToInt(offset != .Immediate)` or `@boolToInt(offset == .Register)` but if you want to be more explicit then using ifs is fine too.

Comment (joachimschmidt557) [reply src-self-hosted/codegen/arm.zig:433]: Hmm yeah, `@boolToInt` is probably the better solution here.

File: test/stage2/test.zig [hunk 2]
```diff
@@ -295,7 +256,7 @@ pub fn addCases(ctx: *TestContext) !void {
             \\    );
             \\    unreachable;
             \\}
-        ,
+            ,
```
Comment (joachimschmidt557) [inline test/stage2/test.zig:259]: `zig fmt` changed this. Will there be any conflicts with other PRs?

Comment (andrewrk) [reply test/stage2/test.zig:259]: I just merged #6293 which I believe reverts this, mind running it again with the latest master?

Comment (joachimschmidt557) [reply test/stage2/test.zig:259]: Will do. Just a sec...

Comment (joachimschmidt557) [reply test/stage2/test.zig]: Done.

**Comment by joachimschmidt557** [comment]
Seems like some conflicts appeared. I'll rebase and resolve those.

**Review by andrewrk — APPROVED** [review]
Really nice work @joachimschmidt557. Thanks for being patient with my big breaking changes.

**Comment by joachimschmidt557** [comment]
@andrewrk Not a problem at all! I'm glad that #6250 was merged.

**Pull request merged**